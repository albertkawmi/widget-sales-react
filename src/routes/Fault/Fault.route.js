import React from 'react'
import Store from '../../components/Store'

// This component hits an endpoint that always returns 500
// in order to demonstrate basic error handling

export default class Fault extends React.Component {
  componentWillMount() {
    this.context.fetchFault()
  }
  render() {
    return (
      <div>
        <h2 className="page-heading">A Faulty Page</h2>
        <p>To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.</p>
      </div>
    )
  }
}

Fault.contextTypes = Store.childContextTypes
