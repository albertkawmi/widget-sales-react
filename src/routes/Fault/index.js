import Fault from './Fault.route'
export default Fault
