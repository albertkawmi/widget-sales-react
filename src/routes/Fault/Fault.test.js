/* eslint-env jest */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Fault from './Fault.route'

Enzyme.configure({ adapter: new Adapter() });

const faultRoute = Enzyme.shallow(
  <Router>
    <Fault />
  </Router>
)

describe('<Fault />', () => {
  it('renders without crashing', () => {
    expect(faultRoute).toBeTruthy()
  })

  it('has not been changed unintentionally', () => {
    expect(faultRoute).toMatchSnapshot()
  })
})
