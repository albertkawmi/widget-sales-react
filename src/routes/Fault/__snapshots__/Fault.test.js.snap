// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fault /> renders A Faulty Page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fault />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="page-heading"
>
          A Faulty Page
</h2>,
        <p>
          To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A Faulty Page",
          "className": "page-heading",
        },
        "ref": null,
        "rendered": "A Faulty Page",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.",
        },
        "ref": null,
        "rendered": "To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="page-heading"
>
            A Faulty Page
</h2>,
          <p>
            To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A Faulty Page",
            "className": "page-heading",
          },
          "ref": null,
          "rendered": "A Faulty Page",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.",
          },
          "ref": null,
          "rendered": "To demonstrate error handling, this page makes a request to an endpoint that eventually returns a 500 error. Once the response comes back, a red error message will appear above and stay on screen until  it's dismissed.",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "fetchFault": [Function],
    },
  },
}
`;
